content/panorama/layout/custom_game/common/override_courier_ui.js

```diff
@@ -0,0 +1,35 @@
+function SelectionCourierUpdate(msg) {
+    var needCourier = msg.newCourier;
+    var selected_entities = GetSelectedEntities();
+    var haveCourierInSelect = false;
```

Comment:
```suggestion
    var haveCourierInSelect = selectedEntities.some(Entities.IsCourier);
```

or

```suggestion
    var haveCourierInSelect = selectedEntities.some(function(e) { return Entities.IsCourier(e) });
```

would replace the loop